community/kernel/src/main/java/org/neo4j/kernel/configuration/ssl/SslPolicyLoader.java

```diff
@@ -0,0 +1,312 @@
+        return policyFactory;
+    }
+
+    /**
+     * Use this for retrieving the SSL policy configured under the specified name.
+     *
+     * @param policyName The name of the SSL policy to be returned.
+     * @return Returns the policy defined under the requested name. If the policy name is null
+     * then the null policy will be returned which means that SSL will not be used. It is up
+     * to each respective SSL policy using component to decide exactly what that means.
+     * @throws IllegalArgumentException If a policy with the supplied name does not exist.
+     */
+    public SslPolicy getPolicy( String policyName )
+    {
+        if ( policyName == null )
+        {
+            return null;
+        }
+        else if ( policyName.equals( LEGACY_POLICY_NAME ) )
+        {
+            return getOrCreateLegacyPolicy();
+        }
+
+        SslPolicy sslPolicy = policies.get( policyName );
+
+        if ( sslPolicy == null )
+        {
+            throw new IllegalArgumentException( format( "Cannot find enabled SSL policy with name '%s' in the configuration", policyName ) );
+        }
+        return sslPolicy;
+    }
+
+    private SslPolicy getOrCreateLegacyPolicy()
+    {
+        if ( legacyPolicy != null )
+        {
+            return legacyPolicy;
+        }
+        legacyPolicy = loadOrCreateLegacyPolicy();
+        return legacyPolicy;
+    }
+
+    private SslPolicy loadOrCreateLegacyPolicy()
+    {
+        File privateKeyFile = config.get( LegacySslPolicyConfig.tls_key_file ).getAbsoluteFile();
+        File certficateFile = config.get( LegacySslPolicyConfig.tls_certificate_file ).getAbsoluteFile();
+
+        if ( !privateKeyFile.exists() && !certficateFile.exists() )
+        {
+            String hostname = config.get( default_advertised_address );
+
+            try
+            {
+                pkiUtils.createSelfSignedCertificate( certficateFile, privateKeyFile, hostname );
+            }
+            catch ( Exception e )
+            {
+                throw new RuntimeException( "Failed to generate private key and certificate", e );
+            }
+        }
+
+        PrivateKey privateKey = loadPrivateKey( privateKeyFile, null );
+        X509Certificate[] keyCertChain = loadCertificateChain( certficateFile );
+
+        return new SslPolicy( privateKey, keyCertChain, null, null, ClientAuth.OPTIONAL, trustAllFactory() );
+    }
+
+    private TrustManagerFactory trustAllFactory()
+    {
+        return InsecureTrustManagerFactory.INSTANCE;
+    }
+
+    private void load( Config config, Log log )
+    {
+        Function<ConfigValues,Stream<String>> enumeration = GroupSettingSupport.enumerate( SslPolicyConfig.class );
+        Set<String> policyNames = config.view( enumeration ).collect( toSet() );
+
+        for ( String policyName : policyNames )
+        {
+            if ( policyName.equals( LEGACY_POLICY_NAME ) )
```

Comment:
I get what you mean now. This is not a rejection of the SSL policy itself (which derives from old settings), but rather just filters out so that one cannot create an SSL policy using the new settings. I will add a test to clarify that.